autofs-5.0.4 - fix memory leak reading ldap master map

From: Ian Kent <raven@themaw.net>

When reading the master map the storage allocated by getting the location
value is not freed after use.
---

 CHANGELOG             |    1 +
 modules/lookup_ldap.c |    2 ++
 2 files changed, 3 insertions(+), 0 deletions(-)


diff --git a/CHANGELOG b/CHANGELOG
index 2be7683..a143294 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -36,6 +36,7 @@
 - fix call restorecon when misc device file doesn't exist.
 - clear rpc client on lookup fail.
 - fix lsb init script header.
+- fix memory leak reading master map.
 
 4/11/2008 autofs-5.0.4
 -----------------------
diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index 5a54b5f..9b1180c 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -1563,6 +1563,8 @@ int lookup_read_master(struct master *master, time_t age, void *context)
 			ldap_value_free(values);
 			goto next;
 		}
+		ldap_value_free(values);
+
 		master_parse_entry(parse_buf, timeout, logging, age);
 next:
 		ldap_value_free(keyValue);
